docs/sdk/src/reference_docs/development_environment_advice.rs

```diff
@@ -0,0 +1,95 @@
+//! # Development Environment Advice
+//!
+//! Large Rust projects are known for sometimes long compile times and sluggish dev tooling, and
+//! polkadot-sdk is no exception.
+//!
+//! This page contains some advice to improve your workflow when using common tooling.
+//!
+//! ## Rust Analyzer Configuration
+//!
+//! [Rust Analyzer](https://rust-analyzer.github.io/) is the defacto [LSP](https://langserver.org/) for Rust. Its default
+//! settings are fine for smaller projects, but not well optimised for polkadot-sdk.
+//!
+//! Below is a suggested configuration in Lua (can use with `neovim/nvim-lspconfig`)
+//! with comments describing the rationale for each setting:
+//!
+//! ```lua
+//! ["rust-analyzer"] = {
+//!   cargo = {
+//!     # Check feature-gated code
+//!     features = "all",
+//!     extraEnv = {
+//!       # Use a seperate target dir for Rust Analyzer. Helpful if you want to use Rust
```

Comment:
```suggestion
//!       # Use a separate target dir for Rust Analyzer. Helpful if you want to use Rust
```

Reply:
some of the settings here might be useful, I have them on, but to be honest I turn ra off most of the time
https://github.com/rust-lang/rust-analyzer/issues/6905#issuecomment-1594331006

Reply:
None of these settings seem useful for performance, but let me know if I missed anything